var/spack/repos/builtin/packages/py-slurm-pipeline/package.py

```diff
@@ -0,0 +1,42 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PySlurmPipeline(PythonPackage):
+    """A Python class for scheduling SLURM jobs"""
+
+    homepage = "https://github.com/acorg/slurm-pipeline"
+    url      = "https://pypi.io/packages/source/s/slurm-pipeline/slurm-pipeline-1.1.13.tar.gz"
+
+    version('1.1.13', 'd1f8c78a64718ec5e2e40ba1b6816017')
+
+    depends_on('py-setuptools', type='build')
+    # using open ranges although requirements*.txt give explicit versions
+    depends_on('py-discover@0.4.0:',   type=('build', 'run'))
```

Comment:
I know `discover` is in the `requirements.txt`, but it isn't actually imported in any file...

Reply:
I'll ask around

Reply:
It looks like `discover` is something used for unit testing. Since the PyPI tarballs don't generally come with unit tests, that explains why it isn't imported anywhere. I think it's safe to drop this dependency.

Reply:
Do we generally don't ship testing-dependencies or should these become out-commented dependencies?

Reply:
We recently added a `test` dependency type so we can add dependencies that are only needed to run unit tests that won't affect the hash. However, in my experience, when you upload something to PyPI it only uploads the bare minimum to install your package, so if we download from PyPI we won't be able to run the unit tests even if we add the dependencies. I'm on the fence as to whether we should encourage downloading from PyPI vs GitHub for this reason.

Reply:
Then I'll remove the dependencies for now and add a comment pointing to this discussion.

Personally I'd prefer github. But that's mostly due to the fact that guessing the correct URL is a pain and I don't completely trust pypi.io, as they are still in beta...

Reply:
#2718 is an attempt to workaround the pypi.io problem and fetch things the way PyPI recommends.